### ISSUE TITLE
Warn but ignore unrecognized manifest keys

### ISSUE THREAD

**Issue by dtolnay** [thepowersgang/mrustc#54]
This is the approach Cargo takes to best support using an old Cargo to compile new crates that use Cargo.toml manifest keys introduced in a later version of Cargo.

I have hit this in two unrecognized keys so far.

```
EXCEPTION: Unknown key `publish` in [package]
```

```
TODO: Unknown manifest section example
Aborted (core dumped)
```

In both cases it would have worked to just ignore the key and keep going.

**Comment by thepowersgang** [comment]
Fixed in 732b1c28b9b23f2c749b3f15905c57754b9b45bf (but not tested on manifests with unknown items)

**Comment by dtolnay** [comment]
Thanks! Confirmed fixed for my crate with `publish` and `example` keys.
